cloudinit/config/cc_ssh.py

```diff
@@ -278,7 +278,7 @@ def handle(
                     if gid != -1:
                         # perform same "sanitize permissions" as sshd-keygen
                         os.chown(keyfile, -1, gid)
-                        os.chmod(keyfile, 0o640)
+                        os.chmod(keyfile, 0o600)
                         os.chmod(keyfile + ".pub", 0o644)
```

Comment:
whereas this is the public key

Reply:
of course but you also have a configuration in the beginning of the file: https://github.com/canonical/cloud-init/blob/9aa2e8a408ce61e2d2ccc1dbd5f9c9b97848d2c9/cloudinit/config/cc_ssh.py#L190

Reply:
And the lines 28x are only valid if ssh-keys attribute is empty. As soon as you provide a key with that attribute the permission got be set as defined in line 189,190 and 191 ;)

Reply:
I found the issue. In the test it was decimal used instead of octal.

I changed it to octal and adapt the value as it should be. Now the pipeline should successfully run.